### PR TITLE
[SRE-98] Migrate to haskell.nix

### PR DESCRIPTION
**Pull request by craigem**
Migrate `cardano-shell` to haskell.nix.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
 steps:
-  - label: 'stack coveralls coverage'
-    command:
-      - 'nix-shell -A runCoveralls'
```
Comment (cleverca22) [inline .buildkite/pipeline.yml:4]: do the new jobs still run coveralls?

Comment (craigem) [reply .buildkite/pipeline.yml:4]: No, I thought I'd removed that :-/

Comment (craigem) [reply .buildkite/pipeline.yml:4]: I'll add it back in.

Comment (craigem) [reply .buildkite/pipeline.yml:4]: I added it back in and later commented it out due to errors in the build that I"m not convinced are Nix related.

**Comment by craigem** [comment]
I believe this PR is ready @iohk-devops . 

* Hydra is failing due to jobs that are also failing in master (see earlier comments on this ticket).
* @cleverca22 asked to keep runCoveralls but I've disabled them for now, as I think the issues with that build are not related o this work.
    
From where I sit now, it's ready for review and merge, unless I've missed some housekeeping.

File: cabal.project [hunk 2]
```diff
@@ -1,4 +1,4 @@
-index-state: 2020-04-01T00:00:00Z
+index-state: 2019-06-20T00:00:00Z
```
Comment (erikd) [inline cabal.project]: What? Why?

Comment (rvl) [reply cabal.project]: Not sure why, but Sam already fixed it in a later commit.

**Comment by sevanspowell** [comment]
The issue I'm currently having is that I can't get cardano-launcher to keep it's source directory (specifically "cardano-launcher/configuration").

Adding "keepSource" hasn't seemed to work.

**Comment by rvl** [comment]
@sevanspowell I'm not sure why `keepSource` hasn't worked. It's what we use in cardano-wallet.
You could also try putting stuff in preCheck. e.g.
```
packages.cardano-launcher.components.tests.cardano-launcher-test.preCheck = ''
  cd $src/cardano-launcher
'';
```

**Comment by sevanspowell** [comment]
https://jira.iohk.io/browse/SRE-125 has been created to track the fixing of the `runCoveralls` tests.

I expect it to be fixed today sometime.

**Pull request merged**